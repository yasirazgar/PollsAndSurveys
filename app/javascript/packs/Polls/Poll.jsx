import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Input from '../Utils/Input'

import './Poll.scss'


// Analyse whether this should be functional and class components
class Poll extends Component {
  answerPoll = () => {
    let alreadySelected = event.target.parentElement.getElementsByClassName("checked");
    if (alreadySelected.length > 0) {
      alreadySelected[0].classList.remove("checked");
    }
    event.target.classList.add('checked');
  }

  optionsList = () => {
    let options = []

    for (let i = 0; i < this.props.options.length; i++) {
      options.push(<li key={i.toString()} onClick={this.answerPoll}>{this.props.options[i]} </li>)
    }
    return options
  }

  render() {

    return (
      <div className="poll">
        <div className="poll__question">
          <h2>{this.props.question}</h2>
        </div>

        <ul className="poll__options">
          {this.optionsList()}
        </ul>
      </div>

    );
  }
}

Poll.propTypes = {
  question: PropTypes.string,
  options: PropTypes.array,
  categories: PropTypes.array
}

export default Poll
