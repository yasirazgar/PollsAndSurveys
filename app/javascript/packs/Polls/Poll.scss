.poll {
  margin-top: 50px;
  box-sizing: border-box;

  ul {
    margin: 0;
    padding: 0;

    /* Style the list items */
    li {
      cursor: pointer;
      position: relative;
      padding: 12px 8px 12px 40px;
      background: #eee !important;
      font-size: 18px;
      transition: 0.2s;

      /* make the list items unselectable */
      -webkit-user-select: none !important;
      -moz-user-select: none !important;
      -ms-user-select: none !important;
      user-select: none !important;
      list-style: none;
    }

    /* Set all odd list items to a different color (zebra-stripes) */
    li:nth-child(odd) {
      background: #f9f9f9 !important;
    }

    /* Darker background-color on hover */
    li:hover {
      background: #ddd;
    }

    /* When clicked on, add a background color and strike out text */
    li.checked {
      background-color: green !important;
      color: black;
    }

    /* Add a "checked" mark when clicked on */
    li.checked::before {
      content: '';
      position: absolute;
      border-color: #fff;
      border-style: solid;
      border-width: 0 2px 2px 0;
      top: 10px;
      left: 16px;
      transform: rotate(45deg);
      height: 15px;
      width: 7px;
    }
  }


  /* Style the close button */
  .close, .add {
    position: absolute;
    right: 0;
    top: 0;
    padding: 12px 16px 12px 16px;

    &:hover {
      background-color: #f44336;
      color: white;
    }
  }
}
