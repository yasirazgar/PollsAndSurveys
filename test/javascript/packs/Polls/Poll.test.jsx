import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Poll from 'packs/Polls/Poll'

const mockStore = configureMockStore();

describe("Poll", () => {
  const poll = {
    poll_id: 1,
    question: 'Fav snake',
    categories: [[3, 'Animals']],
    options: {
      Python: {option_id: 1, percentage: 50.0, selected: true},
      Cobra: {option_id: 2, percentage: 25.0, selected: false},
      Viper: {option_id: 3, percentage: 25.0, selected: false},
      Mamba: {option_id: 4, percentage: 0.0,  selected: false}
    }
  }

  it("renders correctly with poll", () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <Poll poll={poll} />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with currentPoll", () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <Poll currentPoll={poll} />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot();
  });
})
