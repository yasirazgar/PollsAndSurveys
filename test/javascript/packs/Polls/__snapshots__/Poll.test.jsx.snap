// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll renders correctly with currentPoll 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Poll)
    currentPoll={
      Object {
        "categories": Array [
          Array [
            3,
            "Animals",
          ],
        ],
        "options": Object {
          "Cobra": Object {
            "option_id": 2,
            "percentage": 25,
            "selected": false,
          },
          "Mamba": Object {
            "option_id": 4,
            "percentage": 0,
            "selected": false,
          },
          "Python": Object {
            "option_id": 1,
            "percentage": 50,
            "selected": true,
          },
          "Viper": Object {
            "option_id": 3,
            "percentage": 25,
            "selected": false,
          },
        },
        "poll_id": 1,
        "question": "Fav snake",
      }
    }
  />
</ContextProvider>
`;

exports[`Poll renders correctly with poll 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Poll)
    poll={
      Object {
        "categories": Array [
          Array [
            3,
            "Animals",
          ],
        ],
        "options": Object {
          "Cobra": Object {
            "option_id": 2,
            "percentage": 25,
            "selected": false,
          },
          "Mamba": Object {
            "option_id": 4,
            "percentage": 0,
            "selected": false,
          },
          "Python": Object {
            "option_id": 1,
            "percentage": 50,
            "selected": true,
          },
          "Viper": Object {
            "option_id": 3,
            "percentage": 25,
            "selected": false,
          },
        },
        "poll_id": 1,
        "question": "Fav snake",
      }
    }
  />
</ContextProvider>
`;
